db/compaction/compaction.cc

```diff
@@ -991,4 +998,83 @@ int Compaction::EvaluatePenultimateLevel(
   return penultimate_level;
 }
 
+void Compaction::FilterInputsForCompactionIterator() {
```

Comment:
What I had in mind was to mark some compactions as delete compaction https://github.com/facebook/rocksdb/blob/0ca691654f24ff42ee5ca1ddb802419dba3e2c17/db/compaction/compaction.h#L211C8-L211C20. But I think your approach here should be more general. Do we expect use cases that will delete partial range instead of the entire DB?

Reply:
It would be nice for this feature to work e.g. for deleting/replacing a table from a MySQL CF with many tables in the same key space.

Reply:
My understanding is deletion compaction is not a safe option here. For example, if we have a standalone range deletion file on L2, and it's being compacted with a file in L3 and there are data in the same range on lower levels. 

The partial deletion case was something mentioned by Sally folks, although I'm not sure if my understanding of what they mean by partial deletion is correct.